### DISCUSSION TITLE
[Bug Report] What did i do wrong? JMusicBot failed to write new config

### DISCUSSION THREAD

**Discussion by Nazunawa** [jagrosh/MusicBot#818] [Troubleshooting] [answered]
**Describe the bug**
after filling the data when running the JMusicBot-0.3.5.jar on java, it says failed to write new config

**To Reproduce**
1.  right-click the jar file, and open with java
2. Fill the bot token and owner id

**Expected behavior**
idk how this works so idek

**Screenshots**
![image](https://user-images.githubusercontent.com/90824958/133572666-1003f4b0-90d5-49a2-a599-68f76cc92b7c.png)

**Debug Info**
I'm on Windows 10 Pro, JMusicbot version 0.3.5 

**Additional context**
double-clicking the jar file just opens the compressed files, so that's a no

**Comment by MichailiK** [comment, marked as answer]
Could you tell us how you exactly start the MusicBot? Preferably in a video.

**Reply by Nazunawa** [reply]
this is the old method
https://user-images.githubusercontent.com/90824958/133580210-9a9ce89e-7489-4114-8fe1-e3ca6b0643c6.mp4

oh and it did show this after saying that
![image](https://user-images.githubusercontent.com/90824958/133580316-3e1bb878-b717-4013-b4da-460b32669101.png)

**Reply by MichailiK** [reply]
Thank you. 2 Things:
1. You leaked your bot's token in the recording. Reset it in Discord's developer portal.
2. As a workaround, do right click -> Open with -> Choose another app, select java and **make sure you tick the bottom checkbox, in order to always run jar files with java**.

**Reply by Nazunawa** [reply]
weirdly enough, the old method used a config in the system32 and using the new one:

> The fastest way to run that music bot is to do the following:
> Shift + Right click in that folder and then `Open PowerShell window here`
> 
> then run it with the following command:
> `java -Dnogui=true -jar JMusicBot-0.3.5.jar`
 
does use the config in the folder with the jar

**Reply by MichailiK** [reply]
It's caused by Windows Explorer setting the current directory to system32 when using right click -> open with. It should have been fixed a while ago. I will eventually look into this & come up with a fix.

**Reply by Nazunawa** [reply]
> Thank you. 2 Things:
> 
> 1. You leaked your bot's token in the recording. Reset it in Discord's developer portal.
> 2. As a workaround, do right click -> Open with -> Choose another app, select java and **make sure you tick the bottom checkbox, in order to always run jar files with java**.

ok yeah i changed the token

**Reply by Nazunawa** [reply]
![image](https://user-images.githubusercontent.com/90824958/133582072-31212999-9495-4c63-bfd1-18602a565388.png)
this one right

**Reply by MichailiK** [reply]
Yes, be sure to tick the checkbox at the bottom as well.

**Reply by Nazunawa** [reply]
okay thanks for everything
and @DJoepie 🙏

**Comment by DJoepie** [comment]
It's trying to write the config file in system32.
That directory is only writeable by an admin.

Are you running the music bot with admin priveledges or something? It shouldn't create the config file in system32.
You could also manually make the config file by placing a `config.txt` file in the same direcotry as the JMusicBot.
For more information check out the [example config file](https://github.com/jagrosh/MusicBot/wiki/Example-Config).

For further questions you might want to check out the [support server](https://discord.com/invite/0p9LSGoRLu6Pet0k)

**Reply by MichailiK** [reply]
This is an issue that is supposed to have been fixed a while back. Creating new files in system folders is bad practice. This is not supposed to happen.

**Reply by Nazunawa** [reply]
![image](https://user-images.githubusercontent.com/90824958/133574689-1d3e3d19-b20f-4237-9090-ab84f7d8ef05.png)
like this?

**Reply by Nazunawa** [reply]
and no im not running it with admin privileges

**Reply by DJoepie** [reply]
Then you might want to do what Michailik said and upload a video on how you're starting the music bot because it shouldn't do that.
The fastest way to run that music bot is to do the following:
Shift + Right click in that folder and then `Open PowerShell window here`

then run it with the following command:
`java -Dnogui=true -jar JMusicBot-0.3.5.jar`

if that doesn't work, you might want to check if you have Java installed.

**Reply by Nazunawa** [reply]
should i record using the method i tried or the new method

**Reply by DJoepie** [reply]
if the new method doesn't work for you, record the old method.

You might want to record the old method anyways. I don't know if it's useful info for the maintainers to know how you got this bug.

**Comment by Shfloop** [comment]
Hey, I had the same issue and solved it by uninstalling java se runtime environment, and installing java 8.

**Reply by Nazunawa** [reply]
well, I needed Java runtime se(I think) for my Minecraft launcher, and I assumed it was enough

**Comment by SySlain** [comment]
> Then you might want to do what Michailik said and upload a video on how you're starting the music bot because it shouldn't do that. The fastest way to run that music bot is to do the following: Shift + Right click in that folder and then `Open PowerShell window here`
> 
> then run it with the following command: `java -Dnogui=true -jar JMusicBot-0.3.5.jar`
> 
> if that doesn't work, you might want to check if you have Java installed.

Which java version should i have?
